instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -99,8 +102,48 @@ public static boolean isRegistered() {
     return REGISTERED.get();
   }
 
-  private static Driver findDriver(String realUrl) {
-    for (Driver candidate : Collections.list(DriverManager.getDrivers())) {
+  /**
+   * Add driver candidate that wasn't registered against {@link DriverManager}. This is mainly
+   * useful when drivers are initialized at native image build time, and you don't want to postpone
+   * class initialization (f.e. runtime initialization could make driver incompatible with other
+   * systems). Drivers registered with this method have higher priority over those registered
+   * against {@link DriverManager}.
+   *
+   * @param driver {@link Driver} that should be registered
+   */
+  public static void addDriverCandidate(Driver driver) {
+    if (driver != null) {
+      DRIVER_CANDIDATES.add(driver);
+    }
+  }
+
+  /**
+   * Remove driver added via {@link #addDriverCandidate(Driver)}.
+   *
+   * @param driver {@link Driver} that should be unregistered
+   * @return true if the driver was unregistered
+   */
+  public static boolean removeDriverCandidate(Driver driver) {
+    return DRIVER_CANDIDATES.remove(driver);
+  }
+
+  /**
+   * Find driver that accepts {@code realUrl}. Drivers registered against {@link #DRIVER_CANDIDATES}
+   * are preferred over {@link DriverManager} drivers.
+   */
+  static Driver findDriver(String realUrl) {
+    Driver driver = null;
+    if (!DRIVER_CANDIDATES.isEmpty()) {
+      driver = findDriver(realUrl, DRIVER_CANDIDATES);
```

Comment:
it looks like `findDriver` will throw exception if not found, is this the desired behaviour or should it fall back to DriverManager if not found here?

Reply:
My bad, fixed and added test that verifies the scenario. Thanks